test/test_foreach.py

```diff
@@ -353,34 +353,60 @@ def test_int_scalar(self, device, dtype):
             for foreach_bin_op, foreach_bin_op_, torch_bin_op in self.bin_ops:
                 tensors = self._get_test_data(device, dtype, N)
                 scalar = 3
-                expected = [torch_bin_op(t, scalar) for t in tensors]
-
-                res = foreach_bin_op(tensors, scalar)
 
                 if dtype == torch.bool:
+                    if foreach_bin_op == torch._foreach_sub:
```

Comment:
nit: this test is called `test_int_scalar`. That's not very descriptive, it looks like what is being tested here is binary ops where the second argument is an int. Maybe something like `test_tensorlist_scalar_op`?